app/factories/GameFactory.js

```diff
@@ -13,16 +13,16 @@ exports.startGameServer = function(players, gameServerPort, path, map) {
         var max = players.length - 1;
         while (count <= max){
             objToJSON.players[count] = new Object();
-            objToJSON.players[count]["runes"] = new Object();
             objToJSON.players[count].rank = "DIAMOND";
-            objToJSON["players"][count].name = players[count].name;
+            objToJSON["players"][count].name = players[count].username;
             objToJSON["players"][count].champion = dictChamps[players[count].championId];
             objToJSON["players"][count].team = dictTeams[players[count].teamId];
-            objToJSON["players"][count].skin = players[count].skinIndex;
-            objToJSON["players"][count].summoner1 = "HEAL";
-            objToJSON["players"][count].summoner2 = "FLASH";
+            objToJSON["players"][count].skin = players[count].skinId;
+            objToJSON["players"][count].summoner1 = "SummonerHeal";
+            objToJSON["players"][count].summoner2 = "SummonerFlash";
             objToJSON["players"][count].ribbon = 2;
             objToJSON["players"][count].icon = 0;
+            objToJSON.players[count]["runes"] = new Object();
             objToJSON["players"][count]["runes"] = fillRune();
             count++;
         }
```

Comment:
Why is this loop a `while` instead of `for (let i = 0; i < count; i++)`. Also, why do you manually construct an object when this would also work:

    objToJSON.players[count] = {
        rank: "DIAMOND",
        name: players[count].username,
        champion: dictChamps[players[count].championId],
        team: dictTeams[players[count].teamId],
        skin: players[count].skinId,
        summoner1: "SummonerHeal",
        summoner2: "SummonerFlash",
        ribbon: 2,
        icon: 0,
        runes: fillRune()
    };